spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EnvironmentRepositoryConfiguration.java

```diff
@@ -80,8 +78,8 @@ public EnvironmentWatch environmentWatch() {
 
 @Configuration
 @ConditionalOnMissingBean(EnvironmentRepository.class)
+@Profile("!composite")
```

Comment:
You'll have to explain this. Seems counter-intuitive to me.

Reply:
I'm finding that evaluation of `@ConditionalOnMissingBean` is happening before the `EnvironmentRepository`s exist.

Reply:
the order of the `@Import` matters. You've put composite after default.

Reply:
Thanks @spencergibb, that worked.